modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -1366,6 +1366,24 @@ void TFImporter::populateNet(Net dstNet)
 
             connectToAllBlobs(layer_id, dstNet, parsePin(layer.input(0)), id, layer.input_size());
         }
+        else if (type == "MaxPoolGrad")
+        {
+            CV_Assert(layer.input_size() == 3);
+
+            layerParams.set("pool_k_h", 0);
+            layerParams.set("pool_k_w", 0);
+            layerParams.set("pool_stride_h", 0);
+            layerParams.set("pool_stride_w", 0);
+            layerParams.set("pool_pad_h", 0);
+            layerParams.set("pool_pad_w", 0);
```

Comment:
Wait a minute, you already printed text graph with `ksize` and `strides`. So we can use it?

https://github.com/opencv/opencv/pull/14860#issuecomment-505729393

Reply:
No, i do not think so, the shape calculated using ksize and strides may not be correct.

Reply:
Kernels and strides are used in forward pass. To calculate output shape we use `output[2]`.

Reply:
I use input1’s output2(maxpool's index) directly, so I don't need to recalculate it.

Reply:
Oh, yes, I sorry. `MaxUnpoolLayerImpl::forward` just doesn't use them at all.